### PR TITLE
K8SPXC-524 cleanup cluster wide operator correctly after the test

### PR DESCRIPTION
**Pull request by delgod**
[K8SPXC-524](https://jira.percona.com/browse/K8SPXC-524)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | failed |
| storage | failed |
| scaling | failed |
| restore-to-encrypted-cluster | failed |
| self-healing | failed |
| scaling-proxysql | failed |
| big-data | failed |
| smart-update | failed |
| demand-backup | failed |
| self-healing-advanced | failed |
| upgrade | failed |
| scheduled-backup | failed |
| operator-self-healing | failed |
| upgrade-consistency | failed |
| demand-backup-encrypted-with-tls | failed |
| security-context | failed |
| haproxy | failed |
| init-deploy | failed |
| limits | failed |
| monitoring | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| recreate | failed |
| scaling-proxysql | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | failed |
| demand-backup | failed |
| big-data | failed |
| upgrade-consistency | failed |
| storage | failed |
| scheduled-backup | failed |
| smart-update | failed |
| security-context | failed |
| self-healing | failed |
| demand-backup-encrypted-with-tls | failed |
| self-healing-advanced | failed |
| operator-self-healing | failed |
| haproxy | failed |
| init-deploy | failed |
| limits | failed |
| monitoring | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | failed |
| haproxy | failed |
| big-data | failed |
| scaling | failed |
| restore-to-encrypted-cluster | failed |
| smart-update | failed |
| init-deploy | failed |
| demand-backup | failed |
| scaling-proxysql | failed |
| limits | failed |
| scheduled-backup | failed |
| upgrade | failed |
| monitoring | failed |
| storage | failed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |
| self-healing | failed |
| monitoring-2-0 | failed |
| security-context | failed |
| self-healing-advanced | failed |
| affinity | failed |
| operator-self-healing | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| smart-update | failed |
| scaling | failed |
| scaling-proxysql | failed |
| upgrade | failed |
| big-data | failed |
| upgrade-consistency | failed |
| storage | failed |
| haproxy | failed |
| security-context | failed |
| recreate | failed |
| self-healing | failed |
| init-deploy | failed |
| restore-to-encrypted-cluster | failed |
| self-healing-advanced | failed |
| demand-backup | failed |
| limits | failed |
| operator-self-healing | failed |
| monitoring | failed |
| scheduled-backup | failed |
| demand-backup-encrypted-with-tls | failed |
| monitoring-2-0 | failed |
| affinity | failed |
| one-pod | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| scaling | passed |
| haproxy | passed |
| big-data | passed |
| smart-update | failed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| init-deploy | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | failed |

File: e2e-tests/functions [hunk 1]
```diff
@@ -56,7 +56,10 @@ create_namespace() {
     local skip_clean_namespace="$2"
 
     if [[ "${CLEAN_NAMESPACE}" == 1 ]] && [[ -z "${skip_clean_namespace}" ]]; then
-        kubectl get ns | egrep -v "^kube-|^default|Terminating|pxc-operator|openshift|^NAME" | awk '{print$1}' | xargs kubectl delete ns &
+        kubectl_bin get ns \
+            | egrep -v "^kube-|^default|Terminating|pxc-operator|openshift|^NAME" \
+            | awk '{print$1}' \
+            | xargs kubectl delete ns &
```
Comment (cap1984) [inline e2e-tests/functions:62]: does xargs work properly?

File: e2e-tests/functions [hunk 2]
```diff
@@ -183,6 +187,14 @@ create_infra() {
     local ns="$1"
 
     if [ -n "$OPERATOR_NS" ]; then
+        kubectl get pxc --all-namespaces -o wide \
+            | grep -v 'NAMESPACE' \
+            | xargs -L 1 sh -xc 'kubectl patch pxc -n $0 $1 --type=merge -p "{\"metadata\":{\"finalizers\":[]}}"' \
```
Comment (cap1984) [inline e2e-tests/functions:192]: null also fits here

**Review by cap1984 — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| big-data | passed |
| recreate | passed |
| smart-update | failed |
| storage | passed |
| scaling | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| scheduled-backup | passed |
| users | failed |
| demand-backup-encrypted-with-tls | passed |

File: e2e-tests/functions (hunk 1 above)
Comment (delgod) [reply e2e-tests/functions:62]: any reason why not? just check with the command below
```
echo 123 123 | xargs echo xargs-echo: &
```

File: e2e-tests/functions (hunk 2 above)
Comment (delgod) [reply e2e-tests/functions:192]: it is possible to do in different ways, `[]` works well.
Is any reason why we should use `null` here?

**Review by hors — APPROVED** [review]

**Pull request merged**